srtcore/handshake.cpp

```diff
@@ -192,8 +192,7 @@ bool CHandShake::valid()
     if (m_iVersion < CUDT::HS_VERSION_UDT4
             || m_iISN < 0 || m_iISN >= CSeqNo::m_iMaxSeqNo
             || m_iMSS < 32
-            || m_iFlightFlagSize < 2
-            || m_iID >= CUDTUnited::MAX_SOCKET_VAL)
+            || m_iFlightFlagSize < 2)
         return false;
```

Comment:
Let's split this PR into two.
I agree with the removal of this check for socket ID value.
It might be still worth checking the value, but given the check was added recently in #1781 and result in a connection error - can be deleted.

But I am not yet sure about changing `MAX_SOCKET_VAL` and changing random Socket ID generation.

Reply:
Ok, #1817 replaces.

Reply:
The value of `SRTGROUP_MASK` is `1 << 30`, that is 0x4000000. The special values:
* 0 - used in the handshake for connection request
* -1 - invalid socket value or error marker

are considered excluded. Therefore the last 2 bits are reserved (values 0x8000'0000 and 0x4000'0000).

The current version of MAX_SOCKET_VAL is 1 << 29, that is, 0x2000'0000, which is wrong because it excludes also the range between 0x2000'0000 and 0x3FFF'FFFF. The goal of this constant is to exclude only the range between 0x4000'000 and 0x7FFF'FFFF, which are reserved for group IDs. This is required in the socket ID generator that should, in case when decreased to 0, roll back to 0x7FFF'FFFF. But with the current value of 1 << 29 it actually rolls back to 0x3FFF'FFFF. Not that it causes problems, but it contradicts the definitions.

It's not a problem for getting proper IDs for sockets and groups because generation of the group ID relies on generating a regular socket ID and set it the `SRTGROUP_MASK` flag. So, it can potentially generate 0x3FFF'FFFF and as a group ID it will return the value of 0x7FFF'FFFF.